src/components/scenes/existingAccout/ChangeUsernameScene.tsx

```diff
@@ -23,17 +23,20 @@ export const ChangeUsernameScene = (props: ChangeUsernameProps) => {
   const { branding, route } = props
   const { onComplete = () => {} } = useImports()
   const account: EdgeAccount = route.params.account
+  const password = route.params.password
 
   const handleNext = useHandler((newUsername: string) => {
-    onComplete({ username: newUsername })
+    account
+      .changeUsername({ username: newUsername, password })
+      .catch(error => console.error(error))
```

Comment:
Please make `handleNext` `async`, and then just `await` the `account.changeUsername`. Then the button will spin until it's done, and you get error handling for free.